.buildkite/bk.integration.pipeline.yml

```diff
@@ -534,18 +547,54 @@ steps:
             variants:
             - "basic,slim,complete,service,elastic-otel-collector"
             - "wolfi,slim-wolfi,complete-wolfi,elastic-otel-collector-wolfi"
-
             # See the Kubernetes Compatible Versions for Elastic Agent in the Elastic
             # Support Matrix (https://www.elastic.co/support/matrix#matrix_kubernetes) for
             # the rationale on why these specific versions are chosen.
             version:
-              - v1.27.16
-              - v1.28.15
-              - v1.29.14
-              - v1.30.0
-              - v1.31.0
-              - v1.32.0
-              - v1.33.0
+              *k8s_test_versions
+      - label: ":open-pull-request: :kubernetes: {{matrix.version}}:amd64:{{matrix.variants}}"
```

Comment:
The `env`, `command`, `artifact_paths`, `retry`, `agents`, `plugins`, and `matrix.setup.variants` blocks are identical to the previous non-PR step.  Could we use YAML references to reduce the duplication to avoid the risk of drift?

Reply:
the `matrix.setup.variants` are not the same as we cannot parallelize in the same way when testing all k8s versions.

I am not sure that YAML anchors are supported for all the keys you listed, BK docs are very [plugin-focused](https://buildkite.com/docs/pipelines/integrations/plugins/using#using-yaml-anchors-with-plugins)